src/main/java/io/codelex/TestAdvanced/Exercise1/CreditCard.java

```diff
@@ -0,0 +1,29 @@
+package io.codelex.TestAdvanced.Exercise1;
+
+public class CreditCard extends Card {
+
+    private static final int MIN_BALANCE = 100;
+
+    public CreditCard(int number, String fullName, double ccv, double balance) {
+        super(number, fullName, ccv, balance);
+    }
+
+    @Override
+    public void addMoney(double amount) {
+        setBalance(getBalance() + amount);
+
+    }
+
+    @Override
+    public void takeMoney(double amount) throws NotEnoughFundsException {
+        try {
+            setBalance(getBalance() - amount);
+
+            if (getBalance() < MIN_BALANCE) {
+                System.out.println("Warning: Low funds");
+            }
+        } catch (ArithmeticException e) {
```

Comment:
Kāpēc ArithmeticException? Vai kods try blokā to var izmest? Kā arī kods faktiski nemet kļūdu NotEnoughFundsException.
Metodei būtu idejiski jābūt šādai:
```
if (balance < amount) {
   mest kļūdu
}
setBalance(getBalance() - amount);
if (getBalance() < MIN_BALANCE) {
    System.out.println("Warning: Low funds");
}
```

Reply:
Hey, salaboju.